File: API/API/repositorys/UserRepository.vb
Imports System.Data.Entity

Public Class UserRepository
    Implements IUserRepository

    Private ReadOnly _db As AppDbContext

    Public Sub New(db As AppDbContext)
        _db = db
    End Sub

    Public Function GetByEmail(email As String) As User Implements IUserRepository.GetByEmail
        Return _db.Users.FirstOrDefault(Function(u) u.Email = email)
    End Function

    Public Function GetById(id As Guid) As User Implements IUserRepository.GetById
        Return _db.Users.Find(id)
    End Function

    Public Function Exists(email As String) As Boolean Implements IUserRepository.Exists
        Return _db.Users.Any(Function(u) u.Email = email)
    End Function

    Public Sub Create(user As User) Implements IUserRepository.Create
        _db.Users.Add(user)
    End Sub

    Public Sub Save() Implements IUserRepository.Save
        _db.SaveChanges()
    End Sub

    Public Function SearchUsers(keyword As String) As List(Of UserSearchResponse) _
        Implements IUserRepository.SearchUsers

        Dim lower = keyword.Trim().ToLower()
        Return _db.Users.
            Where(Function(u) u.IsActive AndAlso u.Email.ToLower().Contains(lower)).
            OrderBy(Function(u) u.Email).
            Take(10).
            Select(Function(u) New UserSearchResponse With {
                .UserId = u.Id,
                .Email = u.Email
            }).
            ToList()

    End Function

End Class

File: API_v2/Repositorys/UserRepository.cs
using API_v2.Datas;
using API_v2.Models;
using API_v2.Models.DTOs;
using API_v2.Repositorys.IRepositorys;
using Microsoft.EntityFrameworkCore;

namespace API_v2.Repositorys
{
    public class UserRepository : IUserRepository
    {
        private readonly AppDbContext _db;

        public UserRepository(AppDbContext db)
        {
            _db = db;
        }

        public User? GetByEmail(string email)
        {
            return _db.Users.AsNoTracking().FirstOrDefault(u => u.Email.ToLower() == email.ToLower());
        }

        public User? GetById(Guid id)
        {
            return _db.Users.Find(id);
        }

        public void Create(User user)
        {
            _db.Users.Add(user);
        }

        public void Save()
        {
            _db.SaveChanges();
        }

        public List<UserSearchResponse> SearchUsers(string keyword)
        {
            var lower = keyword.Trim().ToLower();
            return _db.Users
                .Where(u => u.IsActive && u.Email.ToLower().Contains(lower))
                .OrderBy(u => u.Email)
                .Take(10)
                .Select(u => new UserSearchResponse
                {
                    UserId = u.Id,
                    Email = u.Email
                })
                .ToList();
        }
    }
}
